plugins/catalog/src/alpha.tsx

```diff
@@ -65,12 +81,203 @@ export const CatalogSearchResultListItemExtension =
+    id: `catalog.filter.${options.id}`,
+    at: 'catalog/filters',
+    inputs: options.inputs ?? {},
+    configSchema: options.configSchema,
+    output: {
+      element: coreExtensionData.reactElement,
+    },
+    factory({ bind, config, source }) {
+      const LazyComponent = React.lazy(() =>
+        options
+          .loader({ config })
+          .then(element => ({ default: () => element })),
+      );
+
+      bind({
+        element: (
+          <ExtensionBoundary source={source}>
+            <React.Suspense fallback={<Progress />}>
+              <LazyComponent />
+            </React.Suspense>
+          </ExtensionBoundary>
+        ),
+      });
+    },
+  });
+}
+
+/** @alpha */
+export const CatalogEntityTagFilter = createCatalogFilterExtension({
+  id: 'entity.tag',
+  loader: async () => {
+    const { EntityTagPicker } = await import('@backstage/plugin-catalog-react');
+    return <EntityTagPicker />;
+  },
+});
+
+/** @alpha */
+export const CatalogEntityKindFilter = createCatalogFilterExtension({
+  id: 'entity.kind',
+  configSchema: createSchemaFromZod(z =>
+    z.object({
+      initialFilter: z.string().default('component'),
+    }),
+  ),
+  loader: async ({ config }) => {
+    const { EntityKindPicker } = await import(
+      '@backstage/plugin-catalog-react'
+    );
+    return <EntityKindPicker initialFilter={config.initialFilter} />;
+  },
+});
+
+/** @alpha */
+export const CatalogEntityTypeFilter = createCatalogFilterExtension({
+  id: 'entity.type',
+  loader: async () => {
+    const { EntityTypePicker } = await import(
+      '@backstage/plugin-catalog-react'
+    );
+    return <EntityTypePicker />;
+  },
+});
+
+/** @alpha */
+export const CatalogEntityOwnerFilter = createCatalogFilterExtension({
+  id: 'entity.mode',
+  configSchema: createSchemaFromZod(z =>
+    z.object({
+      mode: z.enum(['owners-only', 'all']).optional(),
+    }),
+  ),
+  loader: async ({ config }) => {
+    const { EntityOwnerPicker } = await import(
+      '@backstage/plugin-catalog-react'
+    );
+    return <EntityOwnerPicker mode={config.mode} />;
+  },
+});
+
+/** @alpha */
+export const CatalogEntityNamespaceFilter = createCatalogFilterExtension({
+  id: 'entity.namespace',
+  loader: async () => {
+    const { EntityNamespacePicker } = await import(
+      '@backstage/plugin-catalog-react'
+    );
+    return <EntityNamespacePicker />;
+  },
+});
+
+/** @alpha */
+export const CatalogEntityLifecycleFilter = createCatalogFilterExtension({
+  id: 'entity.lifecycle',
+  loader: async () => {
+    const { EntityLifecyclePicker } = await import(
+      '@backstage/plugin-catalog-react'
+    );
+    return <EntityLifecyclePicker />;
+  },
+});
+
+/** @alpha */
+export const CatalogEntityProcessingStatusFilter = createCatalogFilterExtension(
+  {
+    id: 'entity.processing.status',
+    loader: async () => {
+      const { EntityProcessingStatusPicker } = await import(
+        '@backstage/plugin-catalog-react'
+      );
```

Comment:
Feeling that async loading of these is the way to go, but also gonna be a bit interesting to keep an eye on app perf as we start lazy loading more aggressively

Reply:
Thinking now, that maybe just the page should be async because the user might not access it on the first application load, but when the user accesses the page, the filters should be immediately there... I'll convert the filters to be synchronously loaded 🙂

Reply:
Done 🙌🏻 : https://github.com/backstage/backstage/pull/20355/commits/ecaf52aea5076fb5455e2d66c0070910b85867fc

Reply:
Oh no please revert that 😁, this will cause the filters to be eagerly loaded. While the catalog page it lazy these extensions end up having an eager reference to the entire filter component

Reply:
I dropped the commit 😅